src/expr/src/scalar/like_pattern.rs

```diff
@@ -73,3 +147,774 @@ pub fn build_regex(pattern: &str, flags: &str) -> Result<Regex, EvalError> {
+                assert!(self.next.is_none(), "empty suffix in middle of a pattern");
+                return true;
+            }
+            // Use rfind so we perform a greedy capture, like Regex.
+            let mut found = rest.rfind(&self.suffix);
+            loop {
+                match found {
+                    None => return false,
+                    Some(offset) => {
+                        let end = offset + self.suffix.len();
+                        let tmp = &rest[end..];
+                        match &self.next {
+                            None => return tmp.len() == 0,
+                            Some(next) => {
+                                if next.is_match(tmp) {
+                                    return true;
+                                }
+                            }
+                        }
+                        // Didn't match, look for the next rfind.
+                        if offset == 0 {
+                            return false;
+                        }
+                        found = rest[..(end - 1)].rfind(&self.suffix);
+                    }
+                }
+            }
+        }
+        // No string search needed, we just use a prefix match on rest.
+        if !rest.starts_with(&self.suffix) {
+            return false;
+        }
+        rest = &rest[self.suffix.len()..];
+        match &self.next {
+            None => rest.len() == 0,
+            Some(matcher) => matcher.is_match(rest),
+        }
+    }
+}
+
+/// Breaks a LIKE pattern into a chain of sub-patterns.
+fn build_string_matcher(pattern: &str, escape: char) -> Result<StringMatcher, EvalError> {
+    let mut head = StringMatcher {
+        consume: 0,
+        many: false,
+        suffix: String::from(""),
+        next: None,
+    };
+    let mut tail = &mut head;
+    let mut in_wildcard = true;
+    let mut in_escape = false;
+
+    for c in pattern.chars() {
+        match c {
+            c if !in_escape && c == escape => {
+                in_escape = true;
+                in_wildcard = false;
+            }
+            '_' if !in_escape => {
+                if in_wildcard {
+                    tail.consume += 1;
+                } else {
+                    // start a new partial match
+                    tail.next = Some(Box::new(StringMatcher {
+                        consume: 1,
+                        many: false,
+                        suffix: String::from(""),
+                        next: None,
+                    }));
+                    tail = tail.next.as_deref_mut().unwrap();
+                    in_wildcard = true;
+                }
+            }
+            '%' if !in_escape => {
+                if in_wildcard {
+                    tail.many = true;
+                } else {
+                    // start a new partial match
+                    tail.next = Some(Box::new(StringMatcher {
```

Comment:
If you have derived the trait `Default`, you can replace this with
```
let mut next = StringMatcher::default();
next.many = true;
tail.next = Some(Box::new(next));
```

Reply:
Done.